{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "619211bc",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import yaml\n",
    "import json\n",
    "import pandas as pd\n",
    "from pathlib import Path"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "d8f9f37a",
   "metadata": {},
   "outputs": [],
   "source": [
    "cwl_dir = '../cwl/'\n",
    "Path(cwl_dir).mkdir(parents=True, exist_ok=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d9bff122",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "20d326c1",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3707f87d",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a3befc98",
   "metadata": {},
   "outputs": [],
   "source": [
    "template = {\n",
    "    'fq_1': {\n",
    "        'class': 'File',\n",
    "        'path': 'a/file/path'\n",
    "    },\n",
    "    'fq_2': {\n",
    "        'class': 'File',\n",
    "        'path': 'a/file/path'\n",
    "    },\n",
    "    'genome_lib_dir': {\n",
    "        'class': 'Directory',\n",
    "        'path': 'a/file/path'\n",
    "    },\n",
    "    'genome_db': {\n",
    "        'class': 'Directory',\n",
    "        'path': 'a/file/path'\n",
    "    },\n",
    "    'bwts': {\n",
    "        'class': 'Directory',\n",
    "        'path': 'a/file/path'\n",
    "    },\n",
    "    'integrate_executable': {\n",
    "        'class': 'File',\n",
    "        'path': 'a/file/path'\n",
    "    },\n",
    "    'integrate_fasta': {\n",
    "        'class': 'File',\n",
    "        'path': 'a/file/path'\n",
    "    },\n",
    "    'integrate_annotations': {\n",
    "        'class': 'File',\n",
    "        'path': 'a/file/path'\n",
    "    },\n",
    "    'filter_database': {\n",
    "        'class': 'Directory',\n",
    "        'path': 'a/file/path'\n",
    "    },\n",
    "    'fusion_annotator_dir': {\n",
    "        'class': 'Directory',\n",
    "        'path': 'a/file/path'\n",
    "    },\n",
    "    'sample': 'a_string',\n",
    "    'cpu': 'a_int'\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4b1cbe3b",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "044d19de",
   "metadata": {},
   "outputs": [],
   "source": [
    "import re"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "2c6d74cc",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'path/to/a/file/sample.a.woo-boo.T.R1'"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fp = 'path/to/a/file/sample.a.woo-boo.T.R1.fastq'\n",
    "re.sub(r'^(.*).((fq)|(fastq))(.gz)?', r'\\1', fp)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "360e748a",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a33f1ff6",
   "metadata": {},
   "outputs": [],
   "source": [
    "cwl = {\n",
    "    'class': 'CommandLineTool',\n",
    "    'cwlVersion': 'v1.0',\n",
    "    'id': 'adapter_trimming',\n",
    "    'label': 'adapter_trimming',\n",
    "    'baseCommand': ['python', '/pecgs-cnv/cnv/cnv.py'],\n",
    "    'inputs': [\n",
    "        {\n",
    "            'id': 'sample',\n",
    "            'type': 'string',\n",
    "            'inputBinding': {\n",
    "                'position': '1'\n",
    "            }\n",
    "        },\n",
    "        {\n",
    "            'id': 'tumor_bam',\n",
    "            'type': 'File',\n",
    "            'inputBinding': {\n",
    "                'position': '2'\n",
    "            }\n",
    "        },\n",
    "        {\n",
    "            'id': 'normal_bam',\n",
    "            'type': 'File',\n",
    "            'inputBinding': {\n",
    "                'position': '3'\n",
    "            }\n",
    "        },\n",
    "        {\n",
    "            'id': 'reference_dir',\n",
    "            'type': 'Directory',\n",
    "\n",
    "        },\n",
    "        {\n",
    "            'id': 'target_interval_list',\n",
    "            'type': 'File',\n",
    "            'inputBinding': {\n",
    "                'prefix': '--target-interval-list',\n",
    "                'position': '0'\n",
    "            }\n",
    "        },\n",
    "        {\n",
    "            'id': 'common_biallelic',\n",
    "            'type': 'File',\n",
    "            'inputBinding': {\n",
    "                'prefix': '--common-biallelic',\n",
    "                'position': '0'\n",
    "            }\n",
    "        },\n",
    "        {\n",
    "            'id': 'protein_coding_gene',\n",
    "            'type': 'File',\n",
    "            'inputBinding': {\n",
    "                'prefix': '--protein-coding-gene',\n",
    "                'position': '0'\n",
    "            }\n",
    "        },\n",
    "        {\n",
    "            'id': 'pool_of_normals',\n",
    "            'type': 'File',\n",
    "            'inputBinding': {\n",
    "                'prefix': '--pool-of-normals',\n",
    "                'position': '0'\n",
    "            }\n",
    "        },\n",
    "        # needs path to be set so it works on compute1\n",
    "        {\n",
    "            'id': 'environ_PATH',\n",
    "            'type': 'string?',\n",
    "            'default': '/miniconda/envs/cnv/bin:$PATH'\n",
    "        },\n",
    "    ],\n",
    "    'arguments': [\n",
    "        {\n",
    "            'position': 0,\n",
    "            'prefix': '--genome',\n",
    "            'valueFrom': '$(inputs.reference_dir)/GRCh38.d1.vd1.fa'\n",
    "        },\n",
    "        {\n",
    "            'position': 0,\n",
    "            'prefix': '--genome-dict',\n",
    "            'valueFrom': '$(inputs.reference_dir)/GRCh38.d1.vd1.dict'\n",
    "        },\n",
    "        {\n",
    "            'position': 0,\n",
    "            'prefix': '--gene-level-script',\n",
    "            'valueFrom': '/pecgs-cnv/cnv/segment_to_geneLevel.py'\n",
    "        },\n",
    "        {\n",
    "            'position': 0,\n",
    "            'prefix': '--merge-gene-script',\n",
    "            'valueFrom': '/pecgs-cnv/cnv/mergeMultipleFilesToOne.py'\n",
    "        },\n",
    "        {\n",
    "            'position': 0,\n",
    "            'prefix': '--out-dir',\n",
    "            'valueFrom': 'outputs'\n",
    "        },\n",
    "    ],\n",
    "    'outputs': [\n",
    "        {\n",
    "            'id': 'gene_level_cnv',\n",
    "            'type': 'File',\n",
    "            'outputBinding': {'glob': 'outputs/*T.geneLevel.from_seg.cn'}\n",
    "        }\n",
    "    ],\n",
    "    'requirements': [\n",
    "        {\n",
    "            'class': 'DockerRequirement',\n",
    "            'dockerPull': 'estorrs/pecgs_cnv:0.0.1'\n",
    "        },\n",
    "        {\n",
    "            'class': 'ResourceRequirement',\n",
    "            'ramMin': 50000\n",
    "        },\n",
    "        {\n",
    "            'class': 'EnvVarRequirement',\n",
    "            'envDef': {\n",
    "                'PATH': '$(inputs.environ_PATH)',\n",
    "            }\n",
    "        }\n",
    "    ]\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cf58889e",
   "metadata": {},
   "outputs": [],
   "source": [
    "t = {\n",
    "    'class': 'Workflow',\n",
    "    'cwlVersion': 'v1.0',\n",
    "    'id': 'cnv_workflow',\n",
    "    'label': 'cnv_workflow',\n",
    "    'requirements': [{'class': 'ScatterFeatureRequirement'}],\n",
    "    'inputs': [\n",
    "        {\n",
    "            'id': 'sample',\n",
    "            'type': 'string',\n",
    "        },\n",
    "        {\n",
    "            'id': 'tumor_bam',\n",
    "            'type': 'File',\n",
    "        },\n",
    "        {\n",
    "            'id': 'normal_bam',\n",
    "            'type': 'File',\n",
    "        },\n",
    "        {\n",
    "            'id': 'genome',\n",
    "            'type': 'File',\n",
    "        },\n",
    "        {\n",
    "            'id': 'genome_dict',\n",
    "            'type': 'File',\n",
    "        },\n",
    "        {\n",
    "            'id': 'target_interval_list',\n",
    "            'type': 'File',\n",
    "        },\n",
    "        {\n",
    "            'id': 'common_biallelic',\n",
    "            'type': 'File',\n",
    "        },\n",
    "        {\n",
    "            'id': 'protein_coding_gene',\n",
    "            'type': 'File',\n",
    "        },\n",
    "        {\n",
    "            'id': 'pool_of_normals',\n",
    "            'type': 'File',\n",
    "        },\n",
    "    ],\n",
    "    'outputs': [\n",
    "        {\n",
    "            'id': 'gene_level_cnv',\n",
    "            'outputSource': 'cnv/gene_level_cnv',\n",
    "            'type': 'File'\n",
    "        },\n",
    "    ],\n",
    "    'steps': [\n",
    "        {\n",
    "            'id': 'cnv',\n",
    "            'in': [\n",
    "                {'id': 'genome', 'source': 'genome'},\n",
    "                {'id': 'genome_dict', 'source': 'genome_dict'},\n",
    "                {'id': 'target_interval_list', 'source': 'target_interval_list'},\n",
    "                {'id': 'common_biallelic', 'source': 'common_biallelic'},\n",
    "                {'id': 'protein_coding_gene', 'source': 'protein_coding_gene'},\n",
    "                {'id': 'pool_of_normals', 'source': 'pool_of_normals'},\n",
    "                {'id': 'sample', 'source': 'sample'},\n",
    "                {'id': 'normal_bam', 'source': 'stage_normal_bam/output'},\n",
    "                {'id': 'tumor_bam', 'source': 'stage_tumor_bam/output'}],\n",
    "           'label': 'cnv',\n",
    "           'out': [{'id': 'gene_level_cnv'}],\n",
    "           'run': './cnv.cwl'\n",
    "        },\n",
    "        {\n",
    "            'id': 'stage_normal_bam',\n",
    "            'in': [{'id': 'BAM', 'source': 'normal_bam'}],\n",
    "            'label': 'stage_normal_bam',\n",
    "            'out': [{'id': 'output'}],\n",
    "            'run': './stage_bam.cwl'\n",
    "        },\n",
    "        {\n",
    "            'id': 'stage_tumor_bam',\n",
    "            'in': [{'id': 'BAM', 'source': 'tumor_bam'}],\n",
    "            'label': 'stage_tumor_bam',\n",
    "            'out': [{'id': 'output'}],\n",
    "            'run': './stage_bam.cwl'\n",
    "        }\n",
    "    ]\n",
    "}"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
